### DISCUSSION TITLE
[BUG]

### DISCUSSION THREAD

**Discussion by wimdecorte** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1190] [Q&A]
Built the project, Ubuntu 24 LTS on RPI4

The sample video project works fine but the video+audio one does not and errors out with:

```
[DEBUG] [11-06-2024 18:33:25:750.377 GMT] Stream is ready: ExampleStream
[DEBUG] [11-06-2024 18:33:26:243.545 GMT] appsink configured
[INFO ] [11-06-2024 18:33:26:369.404 GMT] Using video device /dev/video1
[INFO ] [11-06-2024 18:33:26:372.366 GMT] Using audio device hw:1,0

**(kvs_gstreamer_audio_video_sample:51265): GLib-GObject-CRITICAL **: 14:33:26.372: g_object_set: assertion 'G_IS_OBJECT (object)' failed
Not all elements could be created:**
[INFO ] [11-06-2024 18:33:26:376.668 GMT] stopKinesisVideoStreamSync(): Synchronously stopping Kinesis Video Stream 0000aaaad3977ce0.
[DEBUG] [11-06-2024 18:33:26:376.803 GMT] streamClosedHandler invoked for upload handle: 18446744073709551615
[INFO ] [11-06-2024 18:33:26:376.879 GMT] Freeing Kinesis Video Stream for ExampleStream
```

Note that it doesn't list the elements that could not be created.

export LD_LIBRARY_PATH= and export GST_PLUGIN_PATH= have been done.
The error happens regardless of using video0 or video1 (I have both the standard RPI camera and a USB camera attached).

**Comment by disa6302** [comment]
@wimdecorte ,

If the sample video is working and not audio, perhaps there is an issue with an audio specific gstreamer element (maybe some RPi specific package that is not installed?).


The error comes from here: 
```
https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/master/samples/kvs_gstreamer_audio_video_sample.cpp#L775-L780

**Reply by wimdecorte** [reply]
Looking at the code, the ":" should probably be removed after the error message.  It implies that a list of missing items will be listed and it doesn't, it just returns.
Ideally though the code should tell us which one of those checks failed so that we can focus troubleshooting a little better.
